### PR TITLE
Enable/Add Felix FV Test Support on arm64

### PR DESCRIPTION
**Pull request by TrevorTaoARM**
Add support for Feliv FV test on arm64:
1. Add ARCH env support in Makefile
2. Add Multi-arch support in Dockerfile.amd64(for which Dockerfile sw links to),
now the building in Makefile uses a single Dockerfile, instead of
Dockerfile.$(arch).
3. Use arch realated image for various images(felix, bpftool)
used in test.
4. Add multi-arch support(ETCD_UNSUPPORTED_ARCH) to Etcd container running environment.

Those changes have been tested on both arm64 and amd64 platforms.
The test results showed it works for both arm64 and amd64 platforms.

Those changes should also work for other architectures(ppcle64, ...), but not tested here.

Signed-off-by: TrevorTaoARM <trevor.tao@arm.com>
Change-Id: Ifca00a88d0164073c5be2c910eced1839a078a28

Signed-off-by: TrevorTaoARM <trevor.tao@arm.com>

## Related issues/PRs
https://github.com/projectcalico/calico/issues/6199

If appropriate, add links to any number of PRs documented by this PR
documents <PR LINK>
-->

## Todos

- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
TBD
```

## Reminder for the reviewer

Make sure that this PR has the correct labels and milestone set.

Every PR needs one `docs-*` label.

- `docs-pr-required`: This change requires a change to the documentation that has not been completed yet.
- `docs-completed`: This change has all necessary documentation completed.
- `docs-not-required`: This change has no user-facing impact and requires no docs.

Every PR needs one `release-note-*` label.

- `release-note-required`: This PR has user-facing changes. Most PRs should have this label.
- `release-note-not-required`: This PR has no user-facing changes.

Other optional labels:

- `cherry-pick-candidate`: This PR should be cherry-picked to an earlier release. For bug fixes only.
- `needs-operator-pr`: This PR is related to install and requires a corresponding change to the operator.

### CODE REVIEW

File: felix/docker-image/Dockerfile.amd64 [hunk 1]
```diff
@@ -22,7 +22,8 @@ FROM ubuntu:focal as wgtool
 RUN apt-get update && \
```
Comment (fasaxc) [inline felix/docker-image/Dockerfile.amd64:22]: If we can now support multiple arches with one dockerfile, let's get rid of the symlink and move this real file into its place (and maybe add a comment saying it's a multi-arch dockerfile now) :tada:

**Review by fasaxc — CHANGES_REQUESTED** [review]
Nice, only thing is that I think it's confusing to have the .amd64 file be a multi-arch dockerfile.  Let's just get rid of the symlink and have one Dockerfile with no suffix :-)

File: felix/docker-image/Dockerfile.amd64 (hunk 1 above)
Comment (TrevorTaoARM) [reply felix/docker-image/Dockerfile.amd64:22]: > If we can now support multiple arches with one dockerfile, let's get rid of the symlink and move this real file into its place (and maybe add a comment saying it's a multi-arch dockerfile now) 🎉

OK, I will update it.

**Comment by odidev** [comment]
Hi Team,

Any updates on PR? When are you planning to merge it?

**Comment by caseydavenport** [comment]
Sorry for the delay here. Kicking off a test run.

**Comment by caseydavenport** [comment]
```
make check-dirtyFailed in 
00:00
The following files are dirty00:00
 felix/fv/infrastructure/etcd.go       | 29 +++++++++++++++--------------00:00
 felix/fv/infrastructure/felix.go      |  4 ++--00:00
 felix/fv/infrastructure/infra_etcd.go |  4 ++--00:00
 felix/fv/infrastructure/infra_k8s.go  | 10 +++++-----00:00
 4 files changed, 24 insertions(+), 23 deletions(-)00:00
lib.Makefile:1116: recipe for target 'check-dirty' failed00:00
make: *** [check-dirty] Error 1
```

Looks like there are some out of date generated files - probably need to run `make generate`

**Comment by caseydavenport** [comment]
Although I didn't think those files were generated...

**Comment by TrevorTaoARM** [comment]
> ```
> make check-dirtyFailed in 
> 00:00
> The following files are dirty00:00
>  felix/fv/infrastructure/etcd.go       | 29 +++++++++++++++--------------00:00
>  felix/fv/infrastructure/felix.go      |  4 ++--00:00
>  felix/fv/infrastructure/infra_etcd.go |  4 ++--00:00
>  felix/fv/infrastructure/infra_k8s.go  | 10 +++++-----00:00
>  4 files changed, 24 insertions(+), 23 deletions(-)00:00
> lib.Makefile:1116: recipe for target 'check-dirty' failed00:00
> make: *** [check-dirty] Error 1
> ```
> 
> Looks like there are some out of date generated files - probably need to run `make generate`

@caseydavenport I will check it some time later. Thank you.

**Comment by lmm** [comment]
Not sure why but it looks like CI is failing due to the dirty files check: https://tigera.semaphoreci.com/jobs/25afeb89-ec03-4da5-bcda-00dc97680a5f

**Comment by odidev** [comment]
Hi Team,

Any updates on PR?

File: felix/fv/infrastructure/etcd.go [hunk 2]
```diff
@@ -15,24 +15,38 @@
...
+	"fmt"
+	"os"
 	log "github.com/sirupsen/logrus"
 
 	"github.com/projectcalico/calico/felix/fv/containers"
 	"github.com/projectcalico/calico/felix/fv/utils"
 )
 
 func RunEtcd() *containers.Container {
+	args := []string{
```
Comment (caseydavenport) [inline felix/fv/infrastructure/etcd.go:27]: You'll need to run `goimports` on this PR as it looks like the formatting isn't quite right.

File: felix/fv/infrastructure/infra_etcd.go [hunk 3]
```diff
@@ -56,6 +56,12 @@ func GetEtcdDatastoreInfra() (*EtcdDatastoreInfra, error) {
 		return nil, errors.New("failed to create etcd container")
 	}
 
+	arch := os.Getenv("ARCH")
```
Comment (caseydavenport) [inline felix/fv/infrastructure/infra_etcd.go:59]: We use this block of code a few times - would be good to throw it in `felix/fv/utils/utils.go` so we don't need to duplicate it so much.

**Review by caseydavenport — CHANGES_REQUESTED** [review]
Seems good to me, a few minor comments on it.

File: felix/fv/infrastructure/etcd.go (hunk 2 above)
Comment (TrevorTaoARM) [reply felix/fv/infrastructure/etcd.go:27]: OK, I will update to it.

File: felix/fv/infrastructure/infra_etcd.go (hunk 3 above)
Comment (TrevorTaoARM) [reply felix/fv/infrastructure/infra_etcd.go:59]: Updated with PR: https://github.com/projectcalico/calico/pull/6989
Should I just abandon this PR?

Comment (TrevorTaoARM) [reply felix/fv/infrastructure/infra_etcd.go:59]: > We use this block of code a few times - would be good to throw it in `felix/fv/utils/utils.go` so we don't need to duplicate it so much.

I added a new func named 'GetSysArch' in utils.go, for your advice.

**Comment by caseydavenport** [comment]
> Updated with PR: https://github.com/projectcalico/calico/pull/6989
> Should I just abandon this PR?

@TrevorTaoARM Yep we can use the other PR.

**Pull request closed without merging**